### PR TITLE
[WIP] Correct BTD with restart

### PR DESCRIPTION
**Pull request by RemiLehe**

### CODE REVIEW

**Comment by n01r** [comment]
I had to remove the second occurrence of 
```
    // When restarting boosted simulations, use correct positions for the moving window
    amrex::Real boosted_moving_window_v = (warpx.moving_window_v - m_beta_boost*PhysConst::c)
                                        / (1._rt - m_beta_boost * warpx.moving_window_v/PhysConst::c);
```
which is from #3689 to make the PR compile.

Original sim step 1001:
```
STEP 1001 starts ...
 Buffer no. 0 lab-frame coords :
 Moving window:
     z (low): -6.823598788e-05
     z (high): -6.396915522e-05
 Z-slice: -6.673592952e-05
 Buffer no. 1 lab-frame coords :
 Moving window:
     z (low): 0.01393176401
     z (high): 0.01393603084
 Z-slice: 0.01393520892
 Buffer no. 2 lab-frame coords :
 Moving window:
     z (low): 0.02793603084
     z (high): 0.02794029768
 Z-slice: 0.02793715377
 Buffer no. 3 lab-frame coords :
 Moving window:
     z (low): 0.04193603084
     z (high): 0.04194029768
 Z-slice: 0.04193909862
 Buffer no. 4 lab-frame coords :
 Moving window:
     z (low): 0.05594029768
     z (high): 0.05594456451
 Z-slice: 0.05594104347
 Buffer no. 5 lab-frame coords :
 Moving window:
     z (low): 0.06994029768
     z (high): 0.06994456451
 Z-slice: 0.06994298832
 Buffer no. 6 lab-frame coords :
 Moving window:
     z (low): 0.08394456451
     z (high): 0.08394883134
 Z-slice: 0.08394493317
 Buffer no. 7 lab-frame coords :
 Moving window:
     z (low): 0.09794456451
     z (high): 0.09794883134
 Z-slice: 0.09794687802
 Buffer no. 8 lab-frame coords :
 Moving window:
     z (low): 0.1119445645
     z (high): 0.1119488313
 Z-slice: 0.1119488229
 Buffer no. 9 lab-frame coords :
 Moving window:
     z (low): 0.1259488313
     z (high): 0.1259530982
 Z-slice: 0.1259507677
 Buffer no. 10 lab-frame coords :
 Moving window:
     z (low): 0.1399488313
     z (high): 0.1399530982
 Z-slice: 0.1399527126
 Buffer no. 11 lab-frame coords :
 Moving window:
     z (low): 0.1539530982
     z (high): 0.153957365
 Z-slice: 0.1539546574
 Buffer no. 12 lab-frame coords :
 Moving window:
     z (low): 0.1679530982
     z (high): 0.167957365
 Z-slice: 0.1679566023
 Buffer no. 13 lab-frame coords :
 Moving window:
     z (low): 0.181957365
     z (high): 0.1819616318
 Z-slice: 0.1819585471
 Buffer no. 14 lab-frame coords :
 Moving window:
     z (low): 0.195957365
     z (high): 0.1959616318
 Z-slice: 0.195960492
 Buffer no. 15 lab-frame coords :
 Moving window:
     z (low): 0.2099616318
     z (high): 0.2099658987
 Z-slice: 0.2099624368
 Buffer no. 16 lab-frame coords :
 Moving window:
     z (low): 0.2239616318
     z (high): 0.2239658987
 Z-slice: 0.2239643817
 Buffer no. 17 lab-frame coords :
 Moving window:
     z (low): 0.2379658987
     z (high): 0.2379701655
 Z-slice: 0.2379663265
 Buffer no. 18 lab-frame coords :
 Moving window:
     z (low): 0.2519658987
     z (high): 0.2519701655
 Z-slice: 0.2519682714
 Buffer no. 19 lab-frame coords :
 Moving window:
     z (low): 0.2659701655
     z (high): 0.2659744323
 Z-slice: 0.2659702162
 Buffer no. 20 lab-frame coords :
 Moving window:
     z (low): 0.2799701655
     z (high): 0.2799744323
 Z-slice: 0.2799721611
 Buffer no. 21 lab-frame coords :
 Moving window:
     z (low): 0.2939701655
     z (high): 0.2939744323
 Z-slice: 0.2939741059
 Buffer no. 22 lab-frame coords :
 Moving window:
     z (low): 0.3079744323
     z (high): 0.3079786992
 Z-slice: 0.3079760508
 Buffer no. 23 lab-frame coords :
 Moving window:
     z (low): 0.3219744323
     z (high): 0.3219786992
 Z-slice: 0.3219779956
 Buffer no. 24 lab-frame coords :
 Moving window:
     z (low): 0.3359786992
     z (high): 0.335982966
 Z-slice: 0.3359799405
 Buffer no. 25 lab-frame coords :
 Moving window:
     z (low): 0.3499786992
     z (high): 0.349982966
 Z-slice: 0.3499818853
 Buffer no. 26 lab-frame coords :
 Moving window:
     z (low): 0.363982966
     z (high): 0.3639872328
 Z-slice: 0.3639838302
 Buffer no. 27 lab-frame coords :
 Moving window:
     z (low): 0.377982966
     z (high): 0.3779872328
 Z-slice: 0.377985775
 Buffer no. 28 lab-frame coords :
 Moving window:
     z (low): 0.3919872328
     z (high): 0.3919914997
 Z-slice: 0.3919877199
 Buffer no. 29 lab-frame coords :
 Moving window:
     z (low): 0.4059872328
     z (high): 0.4059914997
 Z-slice: 0.4059896647
 Buffer no. 30 lab-frame coords :
 Moving window:
     z (low): 0.4199914997
     z (high): 0.4199957665
 Z-slice: 0.4199916096
 Buffer no. 31 lab-frame coords :
 Moving window:
     z (low): 0.4339914997
     z (high): 0.4339957665
 Z-slice: 0.4339935544
 Buffer no. 32 lab-frame coords :
 Moving window:
     z (low): 0.4479914997
     z (high): 0.4479957665
 Z-slice: 0.4479954993
 Buffer no. 33 lab-frame coords :
 Moving window:
     z (low): 0.4619957665
     z (high): 0.4620000333
 Z-slice: 0.4619974441
 Buffer no. 34 lab-frame coords :
 Moving window:
     z (low): 0.4759957665
     z (high): 0.4760000333
 Z-slice: 0.475999389
--- INFO    : Writing buffer 7 of 43 to snapshot 19 to openPMD BTD
             diags/diag_btd
STEP 1001 ends. TIME = 1.335457078e-11 DT = 1.334122955e-14
Evolve time = 51.59037007 s; This step = 0.063971491 s; Avg. per step = 0.05153883124 s
```

Restarted sim from step 1000
```
STEP 1001 starts ...
 Buffer no. 0 lab-frame coords :
 Moving window:
     z (low): -0.003841007686
     z (high): -0.003836740853
 Z-slice: -0.003903510117
```

**Comment by n01r** [comment]
I printed out the `zmax_0` variable. In the original simulation, its value is just `0`. However, for the restarted version, it is `-63.95 µm`.
```
Level 0: dt = 1.334122955e-14 ; dx = 4e-06 ; dy = 4e-06 ; dz = 8.436914022e-05
Buffer no. 0 - zmax_0: -6.394623646e-05
...
Buffer no. 199 - zmax_0: -6.394623646e-05
```

File: Source/Diagnostics/BTDiagnostics.cpp [hunk 1]
```diff
@@ -332,8 +332,15 @@ void
...
+
+    // When restarting boosted simulations, the code below needs to take
+    // into account the fact that the position of the box at the beginning
+    // of the simulation, is not the one that we had at t=0 (because of the moving window)
+    amrex::Real boosted_moving_window_v = (warpx.moving_window_v - m_beta_boost*PhysConst::c)
+                                        / (1._rt - m_beta_boost * warpx.moving_window_v/PhysConst::c);
+
     // Lab-frame time for the i^th snapshot
-    amrex::Real zmax_0 = warpx.Geom(lev).ProbHi(m_moving_window_dir);
+    amrex::Real zmax_0 = warpx.Geom(lev).ProbHi(m_moving_window_dir) - boosted_moving_window_v * warpx.gett_new(0);
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/BTDiagnostics.cpp:343]: I printed the non-restarted and restarted runs and the value of zmax_0 is not the same 
This affects m_t_lab in the line below, and thus affects zslice locations

Comment (RevathiJambunathan) [reply Source/Diagnostics/BTDiagnostics.cpp:343]: thanks for the suggestion. I ll check `boosted_moving_window_v` next

**Comment by RemiLehe** [comment]
This PR is superseded by #3734

**Pull request closed without merging**